### PR TITLE
Mods 29 alternate

### PR DESCRIPTION
**Pull request by gabrielbenavidez**
**This work is still in progress** 

The goal of this commit is to extract AWS secret data instead of lpass when running `db-facts sh`.
The new command should look something like `db-facts sh "cms" --aws`.

Initially I created a method in `lpass.py` to parse the AWS response but I then realized that there is already a python script for this in the `aws_secrets_manager.py`. 

Essentially now the code in `db-info.py` checks if the flag is --aws or --lpass. 
If it's --aws then it converts the lpass secret id into aws secret id. The `db_info_from_secrets_manager` method is then called which returns the DB info from AWS.

Comments/Concerns:
- I commented out some of the Jinja code because I was getting errors when testing (will uncomment in the future).

### CODE REVIEW

**Comment by Brunope** [comment]
Per our conversation earlier, the plan is to continue adding these flags, and add an additional 1password flag. At some point in the future, we will transition this 1password flag to be the default.

File: db_facts/aws_secrets_manager.py [hunk 1]
```diff
@@ -4,6 +4,23 @@
 
 from .db_facts_types import AWSSecret, AWSSecretUsernamePassword
 from .db_type import canonicalize_db_type, db_protocol
+import re
+
+
+"""
+Takes a lastpass secret_id and translates it to the corresponding aws secret_id
+"""
+def translate_secret_id_to_sm(secret_id: str) -> str:
```
Comment (bluelabsbutler) [inline db_facts/aws_secrets_manager.py:13]: [flake8] E302 expected 2 blank lines, found 0

File: db_facts/aws_secrets_manager.py [hunk 2]
```diff
@@ -4,6 +4,23 @@
...
+    cms_secret_pattern = r'(CMS Vertica) (\([a-z]+\)): ([a-z]+)'
+    cms_secret_pattern = re.compile(cms_secret_pattern)
+    if cms_secret_pattern.match(secret_id):
+        match = re.search(cms_secret_pattern, secret_id)
+        stack = match.group(2).replace('(', '').replace(')', '')
+        username = match.group(3)
+        secretsmanager_secret_id = f'{stack}_vertica_{username}_creds'
+    else:
+        raise NotImplementedError("Currently only CMS secret IDs can be translated to secretsmanager")
```
Comment (bluelabsbutler) [inline db_facts/aws_secrets_manager.py:22]: [flake8] E501 line too long (102 > 100 characters)

File: db_facts/db_info.py [hunk 3]
```diff
@@ -4,16 +4,17 @@
...
-    pull_aws_secrets_manager_username_password
+    pull_aws_secrets_manager_username_password,
+    translate_secret_id_to_sm
 )
 from .db_facts_types import DBConfig, DBCLIConfig, DBFacts, DBName
 from .db_config import db_config
 
 
-def db(db_name: DBName, dbcli_config: DBCLIConfig = None) -> DBFacts:
+def db(db_name: DBName, dbcli_config: DBCLIConfig = None, secret_type:str=None) -> DBFacts:
```
Comment (bluelabsbutler) [inline db_facts/db_info.py:17]: [flake8] E231 missing whitespace after '

Comment (bluelabsbutler) [inline db_facts/db_info.py:17]: [flake8] E252 missing whitespace around parameter equals

File: db_facts/db_info.py [hunk 4]
```diff
@@ -36,18 +37,18 @@ def db(db_name: DBName, dbcli_config: DBCLIConfig = None) -> DBFacts:
...
-        else:
-            db_info[k] = template(config[k], jinja_context)
+    # jinja_context = pull_jinja_context(db_name, config, dbcli_config)
+    # for k in config.keys():
+    #     if k == 'jinja_context_name':
+    #         pass
+    #     elif k == 'exports':
+    #         db_info['exports'].update({
+    #             template(exportname,
+    #                      jinja_context): template_any(config['exports'][exportname], jinja_context)
```
Comment (bluelabsbutler) [inline db_facts/db_info.py:47]: [flake8] E501 line too long (101 > 100 characters)

File: db_facts/lpass.py [hunk 5]
```diff
@@ -1,6 +1,11 @@
+import sys
```
Comment (bluelabsbutler) [inline db_facts/lpass.py:1]: [flake8] F401 'sys' imported but unused

File: db_facts/runner.py [hunk 6]
```diff
@@ -67,6 +67,13 @@ def run(self, argv: List[str]) -> int:
                                    help=('Friendly name of database '
                                          '(e.g., "redshift", "dmv", '
                                          '"abc-dev-dbadmin")'))
+            group = sh_parser.add_mutually_exclusive_group()
+            group.add_argument('--lpass', action='store_true',
+                               help=('type of secret service to pull from '
+                                         '(e.g., "--lpass"'))
```
Comment (bluelabsbutler) [inline db_facts/runner.py:73]: [flake8] E127 continuation line over-indented for visual indent

File: db_facts/runner.py [hunk 7]
```diff
@@ -67,6 +67,13 @@ def run(self, argv: List[str]) -> int:
...
+            group.add_argument('--aws', action='store_true',
+                               help=('type of secret service to pull from '
+                                         '(e.g., "--aws"'))
```
Comment (bluelabsbutler) [inline db_facts/runner.py:76]: [flake8] E127 continuation line over-indented for visual indent

**Pull request closed without merging**